### PR TITLE
Replace the unmaintained eui48 crate with the macaddr crate

### PR DESCRIPTION
**Pull request by iluminat23**
- Replace the unmaintained eui48 crate with the macaddr crate
- make the repo REUSE compliant
- update the Cargo.toml
- strip release builds by default

### CODE REVIEW

File: .reuse/dep5 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Source: https://github.com/RevolutionPi/revpi-hat-eeprom
+
+Files: Cargo.lock
+Copyright: KUNBUS GmbH
```
Comment (tboehler1) [inline .reuse/dep5:5]: I'm surprised there is not copyright year here, but checking REUSE compliance with the tool seems to be okay. I guess the year is not required?

**Review by tboehler1 — CHANGES_REQUESTED** [review]
There are multiple files where the copyright year is only "2022" and others where it's "2022-2023". This should be "2022-2023" everywhere imo

File: .reuse/dep5 (hunk 1 above)
Comment (iluminat23) [reply .reuse/dep5:5]: I dropped the year, as the file is generated and updating the year is a useless chore ;)

**Comment by iluminat23** [comment]
> There are multiple files where the copyright year is only "2022" and others where it's "2022-2023". This should be "2022-2023" everywhere imo

The other files have not been changed in 2023, and thus have no newer copyright. Otherwise we wouldn't need to place the header in every file.

**Review by tboehler1 — APPROVED** [review]

**Pull request merged**